android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/MetadataResolver.java

```diff
@@ -90,13 +96,13 @@ public static void findUrlMetadata(final Context context, final MetadataResolver
    *
    * @param url The url for which to request data
    */
-  private static void requestUrlMetadata(String url) {
+  private static void requestUrlMetadata(String url, int txPower, int rssi) {
```

Comment:
What about making this three parameters an object. It seems you use all of them together all the time.